src/components/Nav/Nav.Props.ts

```diff
@@ -78,17 +78,21 @@ export interface INavLink {
   isExpanded?: boolean;
 
   /**
-   * Any additional properties to apply to the rendered links.
+   * (Optional) Aria label for nav link
    */
-  [propertyName: string]: any;
+  ariaLabel?: string;
 
   /**
-   * (Optional) Aria label for nav link
+   * (Optional) Meta info for the link, does not involving rendering.
    */
-  ariaLabel?: string;
+  key?: string;
 
   /**
-   * (Optional) Title for nav link
+   * (Optional) title for tooltip or description
    */
   title?: string;
+  /**
+   * Any additional properties to apply to the rendered links.
+   */
+  [propertyName: string]: any;
```

Comment:
Found out this property always there if you see your initial copy.  But it is not at the end.  I only moved it to the end as standard way.